aspnetcore/fundamentals/logging/loggermessage.md

```diff
@@ -0,0 +1,171 @@
+---
+title: High-performance logging with LoggerMessage in ASP.NET Core
+author: guardrex
+description: Learn how to use LoggerMessage features to create cacheable delegates that require fewer object allocations than logger extension methods for high-performance logging scenarios.
+ms.author: riande
+manager: wpickett
+ms.date: 10/31/2017
+ms.topic: article
+ms.assetid: 1725ffe9-b6be-4fbb-af35-c8b9363e2292
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: fundamentals/logging/loggermessage
+---
+# High-performance logging with LoggerMessage in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+[LoggerMessage](/dotnet/api/microsoft.extensions.logging.loggermessage) features create cacheable delegates that require fewer object allocations than [logger extension methods](/dotnet/api/Microsoft.Extensions.Logging.LoggerExtensions), such as `LogInformation`, `LogDebug`, and `LogError`. For high-performance logging scenarios, use the `LoggerMessage` pattern.
+
+Logger extension methods require "boxing" (converting) value types, such as `int`, into `object` types and back. The `LoggerMessage` pattern avoids boxing by using static `Action` fields to define logging messages for strongly-typed extension methods. 
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/logging/loggermessage/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+The sample app demonstrates `LoggerMessage` features with a simple quote tracking database. The user adds, deletes, and deletes all quotes from an in-memory database. As these operations occur, log messages are generated using static extension methods that trigger logging `Action`s.
+
+## LoggerMessage.Define
+
+[Define(LogLevel, EventId, String)](/dotnet/api/microsoft.extensions.logging.loggermessage.define) creates an `Action` delegate for logging a message. `Define` overloads permit passing up to six type parameters to a named format string (template).
+
+The string provided to the `Define` method is a template and not an interpolated string. Placeholders are filled in the order that the types are specified. Placeholder names in the template should be descriptive. The sample app uses all capital-letter placeholder names to distinguish them from any type of variable.
```

Comment:
It might be worth mentioning why the placeholder names should be descriptive. And consistent across similar log messages.

Reply:
Is the reason merely because they're more descriptive? ... Are they used outside of being a visual aid?

Reply:
btw -- I thought it might be a good idea to capitalize to distinguish them from any var or prop naming. However, that isn't what the dev team is doing. It's Pascal case. Do you think it's best if we go with Pascal case placeholder names here?

Reply:
:duck: says ... *yes* ... I'm converting to Pascal case now. I like that idea better.